external/kafka-0-10-sql/src/main/scala/org/apache/spark/sql/kafka010/KafkaStreamWriter.scala

```diff
@@ -116,3 +133,66 @@ class KafkaStreamDataWriter(
     }
   }
 }
+
+private[kafka010] case class KafkaWriterCustomMetrics(
+    minOffset: KafkaSourceOffset,
+    maxOffset: KafkaSourceOffset) extends CustomMetrics {
+  override def json(): String = {
+    val jsonVal = ("minOffset" -> parse(minOffset.json)) ~
+      ("maxOffset" -> parse(maxOffset.json))
+    compact(render(jsonVal))
+  }
+
+  override def toString: String = json()
+}
+
+private[kafka010] object KafkaWriterCustomMetrics {
+
+  import Math.{min, max}
+
+  def apply(messages: Array[WriterCommitMessage]): KafkaWriterCustomMetrics = {
+    val minMax = collate(messages)
+    KafkaWriterCustomMetrics(minMax._1, minMax._2)
+  }
+
+  private def collate(messages: Array[WriterCommitMessage]):
```

Comment:
good to leave some comment on what this does. It seems to be computing the min/max offset per partition? If so choosing an apt name for that function would make it clearer.

Reply:
Thanks, I will rename to something with minMax.